src/figuresClasses.ts

```diff
@@ -1,19 +1,109 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  color: Color;
+
+  sideA: number;
+
+  sideB: number;
+
+  sideC: number;
+
+  constructor(public color: string, public sideA: number,
+    public sideB: number, public sideC: number) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw Error('All sides must be greater then 0');
+    }
+
+    const sides = [sideA, sideB, sideC].sort((a, b) => a - b);
+
+    if (sides[0] + sides[1] <= sides[2]) {
+      throw Error('Invalid triangle sides');
+    }
+
+    this.color = color;
+
+    this.sideA = sideA;
+
+    this.sideB = sideB;
+
+    this.sideC = sideC;
```

Comment:
you can simplify it using the [Typescript Constructor Shorthand](https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd)